### PR TITLE
bazel: update to v5.0.0

### PR DESCRIPTION
**Pull request by kfaseela**
#18718

Depends on https://github.com/envoyproxy/envoy-build-tools/pull/157

Signed-off-by: Faseela K <faseela.k@est.tech>

### CODE REVIEW

File: .bazelrc [hunk 1]
```diff
@@ -18,6 +18,8 @@ build --incompatible_strict_action_env
 build --host_force_python=PY3
 build --host_javabase=@bazel_tools//tools/jdk:remote_jdk11
 build --javabase=@bazel_tools//tools/jdk:remote_jdk11
+build --java_runtime_version=remotejdk_11
```
Comment (moderation) [inline .bazelrc:21]: I'm not particularly familiar with the Java ecosystem but know that 17 is the latest version. Should we upgrade to `remote_jdk17` as per https://docs.bazel.build/versions/main/user-manual.html#flag--java_runtime_version. If so probably need to adjust https://github.com/envoyproxy/envoy/blob/main/bazel/BUILD#L619-L620 as well (and maybe the Kafka extension)

Comment (phlax) [reply .bazelrc:21]: could we change java separate to bazel - im just thinking less complexity, and easier to revert etc

**Comment by phlax** [comment]
just tracking through trying to figure the couple of errors

one is the same error that @keith hit previously https://github.com/envoyproxy/envoy/issues/18718#issuecomment-949992275

the other is a problem with it not finding `ld` - a bit beyond my understanding of how the toolchains work, but i think the issue is the change here https://github.com/envoyproxy/envoy-build-tools/commit/8e4e3825cb21f1197721240841b2ff8c1d573a72#diff-7d042474e6c30aa209b372e1391823e65a962c03318c945d049e8d40fc272e1dR115 - not sure tho

**Comment by phlax** [comment]
im not certain - but i think that https://github.com/envoyproxy/envoy-build-tools/pull/161 can fix the ld problem

File: .bazelrc (hunk 1 above)
Comment (moderation) [reply .bazelrc:21]: Fair enough

**Comment by phlax** [comment]
@htuch @adisuissa @keith it looks like the only remaining blocker to upgrading to bazel 5 is the pgv issue

Not sure exactly why but it cant find the compiled protobuf files

```
/pgv_test: error while loading shared libraries: libexternal_Senvoy_Uapi_Senvoy_Sconfig_Sbootstrap_Sv3_Slibpkg_Ucc_Uproto.so: cannot open shared object file: No such file or directory
================================================================================
```

any clues on this one ?

**Comment by phlax** [comment]
re the pgv/api errors, and on the basis that the fail seems only to occur in CI, im guessing that the it must be RBE/toolchain related

at the same time the fail was happening before we upgraded the toolchains, so it seems to be something related to how bazel 5 loads external deps (for RBE)

**Comment by keith** [comment]
Can we try disabling remote execution for this CI job / action specifically, I bet this is this issue: https://github.com/bazelbuild/bazel/issues/13819

**Comment by keith** [comment]
For just this test we could add `tags = ["no-remote-exec"]` and see if any other tests failed the same way

**Comment by phlax** [comment]
> ..., I bet this is this issue: 

the issue looks correct in terms of symptoms, but we werent having this problem before bazel 5 and that ticket has been around a while, and refers to 4.1, so im wondering what changed that made it no longer work

**Comment by keith** [comment]
Yea good question, there are some open fixes for it though, so hopefully we won't have to live with any workaround for long

File: api/test/build/BUILD [hunk 2]
```diff
@@ -5,6 +5,7 @@ licenses(["notice"])  # Apache 2
 api_cc_test(
     name = "build_test",
     srcs = ["build_test.cc"],
+    linkstatic = True,
```
Comment (phlax) [inline api/test/build/BUILD:8]: im wondering if we should add a comment with the upstream issue that requires this

File: WORKSPACE [hunk 3]
```diff
@@ -19,3 +19,7 @@ envoy_dependencies_extra()
 load("//bazel:dependency_imports.bzl", "envoy_dependency_imports")
 
 envoy_dependency_imports()
+
+load("@bazel_skylib//lib:versions.bzl", "versions")
```
Comment (phlax) [inline WORKSPACE:23]: @keith was this part necessary ?

if it is im wondering whether it should be in `dependency_imports.bzl` or whether its more explicit in `WORKSPACE`

Comment (keith) [reply WORKSPACE:23]: not required, i was just using it as a safety net to make sure that CI was respecting the updated config and using the right bazel version

**Review by keith — DISMISSED** [review]

**Review by phlax — APPROVED** [review]
lgtm, thanks @kfaseela and @all 

ill leave to assigned maintainer to land, thanks again!

**Review by wrowe — APPROVED** [review]
LGTM. Landing it. We have the entire week to sort out any collateral dmg.

**Comment by keith** [comment]
Great work, thanks for doing this!

**Pull request merged**